client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -132,7 +158,7 @@ export default React.createClass( {
 			name,
 			ref: name,
 			additionalClasses: 'checkout-field',
-			value: formState.getFieldValue( this.state.form, name ),
+			value: formState.getFieldValue( this.state.form, name ) || '',
```

Comment:
Fair enough - I was thinking about primitive/native fields here, but more complex components might indeed need additional logic to handle such cases correctly 👍